src/cascadia/TerminalApp/CascadiaSettings.cpp

```diff
@@ -151,6 +151,55 @@ void CascadiaSettings::_CreateDefaultProfiles()
     _profiles.emplace_back(powershellProfile);
     _profiles.emplace_back(cmdProfile);
     
+    // Add all WSL distros from: HKCU\Software\Microsoft\Windows\CurrentVersion\Lxss
+    Profile WSLDistro{};
+    HKEY hLXSSKey;

```

Comment:
To ensure the lifetime of this key handle is managed appropriately, be sure to use the [wil\resource.h helper types](https://github.com/microsoft/wil/blob/master/include/wil/resource.h):

wil::unique_hkey hLXSSKey;

This ensures that if execution exits this scope for any reason the resource is appropriately freed.  This also means that the manual call to RegCloseKey on 201 becomes unnecessary.

Reply:
Updated.